spring-cloud-kubernetes-client-discovery/src/main/java/org/springframework/cloud/kubernetes/client/discovery/ConditionalOnSelectiveNamespacesMissing.java

```diff
@@ -0,0 +1,61 @@
+/*
```

Comment:
the entire "dance" in this PR is around these two classes. I'll try to explain.

Until now, when you wanted to discovery services using k8s-client, what we needed under the hood was two `Lister`s : one for `V1Service` and one for `V1Endpoints` and that is what is have been creating successfully. 

When you think that now we want to provide support for selective namespaces, like `[namespace-a, namespace-b]`, all we need is a `List<Lister<V1Service>>` and `List<Lister<V1Endpoints>>`. 

So my idea here is this: if users have said that they want to discover services in some namespaces, we should "catch" that and use that information to create proper beans. This is what these classes are for.

So this one : `ConditionalOnSelectiveNamespacesMissing` is active when user have _not_ selected some namespaces - so everything should work as before. On the other hand when `ConditionalOnSelectiveNamespacesPresent` is active, we defer to a different auto-configuration (`KubernetesClientInformerSelectiveNamespacesAutoConfiguration`) that creates lists of listers.

Reply:
I like it!